### PR TITLE
Module split PR4

### PR DESCRIPTION
**Pull request by Pessimistress**
#### Background
Module split for #1629

#### Change List
- Update aliases system
- Update webpack module resolve
- Correct imports in examples and tests

### CODE REVIEW

File: aliases.js [hunk 1]
```diff
@@ -21,20 +21,37 @@
...
-  'deck.gl': path.resolve(__dirname, './src'),
-  'deck.gl-layers': path.resolve(__dirname, './src/experimental-layers/src'),
-  'deck.gl-test-utils': path.resolve(__dirname, './src/test-utils/src')
-};
+const MODULE_PATH = resolve(__dirname, './modules');
 
-if (module.require) {
-  module.require('reify');
+// Get all module info
+const MODULES = {};
```
Comment (ibgreen) [inline aliases.js]: This is an intimidating start to the file. Things that can help:
* Create some well named function for this.
* Place that function at the end of the file, after a `// HELPER FUNCTIONS` comment, just calling it here.
* spend some time on improving comments, e.g`Get all module info` doesn't quite get me started on the right track. `Build a map containing the contents of 'package.json' for all deck.gl modules` might be better.

File: aliases.js [hunk 2]
```diff
@@ -21,20 +21,37 @@
...
+function getAliases(mode = 'src') {
+  const aliases = {};
+
+  for (const moduleName in MODULES) {
+    const subPath = mode === 'src' ? 'src' : MODULES[moduleName].main.replace('/index.js', '');
+    aliases[moduleName] = resolve(__dirname, 'node_modules', moduleName, subPath);
+  }
+
+  return Object.assign({
+    // TODO - remove when core and core-layers are separated
```
Comment (ibgreen) [inline aliases.js]: Might be good to mention the importance of order here as it already bit us here and has before in other places.

File: aliases.js [hunk 3]
```diff
@@ -21,20 +21,37 @@
...
+    'deck.gl/core': resolve(aliases['@deck.gl/core'], 'core'),
+    'deck.gl/core-layers': resolve(aliases['@deck.gl/core'], 'core-layers'),
```
Comment (ibgreen) [inline aliases.js]: Do we have a lot of references to this?

If possible, I'd prefer if we could postpone the decision on what to name the layer module to next release.

* As an example we might have several "core" layer modules by then, making this name redundant.
* Or we might just want to call it `@deck.gl/layers` (currently my personal preference).

File: examples/layer-browser/src/examples/experimental-layers.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import {MeshLayer, PathOutlineLayer, PathMarkerLayer, AdvancedTextLayer} from 'deck.gl-layers';
 
 // TODO: remove hard path once deck.gl-layers published with GPUScreenGridLayer
-import GPUScreenGridLayer from '../../../../src/experimental-layers/src/gpu-screen-grid-layer/gpu-screen-grid-layer';
+import GPUScreenGridLayer from '@deck.gl/experimental-layers/gpu-screen-grid-layer/gpu-screen-grid-layer';
```
Comment (ibgreen) [inline examples/layer-browser/src/examples/experimental-layers.js]: Why are we importing it with a long path like this and not as `import {GPUScreenGridLayer} from '@deck.gl/experimental-layers;`

The whole point of `experimental-layers` is to export "experimental layers".

File: examples/webpack.config.local.js [hunk 5]
```diff
@@ -43,7 +39,7 @@ function makeLocalDevConfig(EXAMPLE_DIR = LIB_DIR) {
         'viewport-mercator-project': resolve(LIB_DIR, './node_modules/viewport-mercator-project'),
```
Comment (ibgreen) [inline examples/webpack.config.local.js]: Nit: probe.gl, mjolnir.js, ???

File: package.json [hunk 6]
```diff
@@ -29,7 +29,7 @@
     "publish-beta": "learna run build && npm run test && npm run test-dist && scripts/publish.sh beta",
     "test": "npm run lint && npm run test-node && npm run test-browser && npm run test-render-react",
     "test-fast": "npm run test-node",
-    "test-ci": "node test/start.js test-ci && npm run build && npm run collect-metrics",
+    "test-ci": "node test/start.js test-ci && npm run collect-metrics",
```
Comment (ibgreen) [inline package.json]: Does this ensure there is a build to measure, or how does that happen?

File: test/bench/gpu-grid-aggregator.bench.js [hunk 7]
```diff
@@ -19,9 +19,9 @@
 // THE SOFTWARE.
 
 /* eslint-disable no-console, no-invalid-this */
-// TODO: remove hard path once deck.gl-layers published with GPUScreenGridLayer
-import GPUGridAggregator from '../../src/experimental-layers/src/utils/gpu-grid-aggregator';
-import {gl} from 'deck.gl-test-utils';
+// TODO: remove hard path once @deck.gl/experimental-layers published with GPUScreenGridLayer
+import GPUGridAggregator from '@deck.gl/experimental-layers/utils/gpu-grid-aggregator';
```
Comment (ibgreen) [inline test/bench/gpu-grid-aggregator.bench.js]: Nit: Can we remove the hard path now?

File: test/start.js [hunk 8]
```diff
@@ -21,7 +21,7 @@ switch (mode) {
 
   case 'test-dist':
     // Load deck.gl itself from the dist folder
-    moduleAlias.addAlias('deck.gl', path.resolve('./dist'));
+    moduleAlias.addAliases(getAliases('dist'));
```
Comment (ibgreen) [inline test/start.js]: Neat!

File: aliases.js (hunk 3 above)
Comment (Pessimistress) [reply aliases.js]: These are just used internally in tests. Note the lack of `@` at the beginning.

File: package.json (hunk 6 above)
Comment (Pessimistress) [reply package.json]: I'm putting it back

**Pull request merged**